transformer_engine/common/common.h

```diff
@@ -169,34 +169,52 @@ struct Tensor {
   }
 
   size_t dim() const {
-    if (!has_data() && has_columnwise_data()) {
+    // Check whether a tensor shape matches an uninitialized tensor
+    auto is_shape_trivial = [](const std::vector<size_t> &shape) -> bool {
+      return shape.size() == 1 && shape[0] == 0;
+    };
+
+    // Choose data buffer based on whether it is initialized
+    // Note: Logically each tensor format interprets its data
+    // differently, but for simplicity we assume they all use row-wise
+    // and column-wise data similarly.
+    bool use_columnwise_shape = false;
+    if (data.dptr != nullptr) {
+      use_columnwise_shape = false;
+    } else if (columnwise_data.dptr != nullptr) {
+      use_columnwise_shape = true;
+    } else if (!is_shape_trivial(data.shape)) {
+      use_columnwise_shape = false;
+    } else if (!is_shape_trivial(columnwise_data.shape)) {
+      use_columnwise_shape = true;
+    }
```

Comment:
And so this is a change in behavior, since now if all of these fail (and so there is no data nor columnwise data and both shapes are trivial) then you end up with dim = 1 instead of 0. Is that expected?